### PR TITLE
[K12] Handle Deletion of Grade Enrollments

### LINKED ISSUE #169: k12: Handle Deletion of Grade Enrollments
ENHANCEMENT REQUEST

What is the current behavior?

When all of a Contact's Grade Enrollments have been deleted the "Current Grade Level" field on Contact Details still shows whichever Grade Level was last marked as Contact.

What is the expected behavior?

When a Contact has no Grade Enrollment Records marked as "Current", the Current Grade Level field on Contact should be changed to 'Not Applicable'.

Steps:

Create Contact, Name = Marcus Jefferson
Create Multiple Grade Enrollment Records, with Grade Level & Status :
--> First Grade, Former
--> Second Grade, Former
--> Third Grade, Current
Note that the "Current Grade Level" on the Contact Details now says "Third Grade."
Delete the Third Grade, Current Grade Enrollment 
--> Actual Result : Marcus's "Current Grade Level" still says "Third Grade"
--> Expected Result : Marcus's "Current Grade Level" is blank, since he is not currently in that school. (Alternatively, says "N/A".)
What is the motivation / use case for changing the behavior?

So that school personal can accurately use the Contact Details to know if a student is enrolled in a grade at their school
Keeping Data in Sync
Less work trying to maintain data in 2 places
Acceptance Criteria

Given: A Contact with Grade Enrollment Records
When: None are marked "Current"
Then: The "Current Grade Level" field on that Contact's details now changed to 'Not Applicable'.

### PR DESCRIPTION
**Pull request by skosgi1**
# Critical Changes

# Changes
## Updates to the Current Grade Level Field on Contacts
We've updated the GRER_SyncGradeLevel_TDTM class to ensure that the Current Grade Level field on Contact is updated when a Grade Enrollment record with a "Current" status that matches the Contact's Current Grade Level is deleted. Now, you'll see the following behaviors:

- If no other Grade Enrollment record associated with this Contact has a status of "Current," the Current Grade Level field on Contact is updated to "Not Applicable."
- If one additional Grade Enrollment record associated with this Contact has a status of "Current," the Current Grade Level field on Contact is updated to match the grade level of that Grade Enrollment record.
- If more than one remaining Grade Enrollment record associated with this Contact has a status of "Current," users receive an error message if they try to change the status of a current Grade Enrollment record with a Grade Level that matches the Current Grade Level field on the associated Contact.
**Note:** We recommend that you do not disable this trigger handler.

# Issues Closed
Closes #169 

# New Metadata

# Deleted Metadata

# Testing Notes
1. Scenario One:
    a. Create a Contact
    b. Create a GE record with above Contact and Status as ‘Current’ with ‘Grade Level’ as ‘Prekindergarten’
    c. Ensure Current Grade Level on Contact is populated as ‘Prekindergarten’
    d. Delete the GE record
    e. Ensure Current Grade Level on Contact is changed to ‘Not Applicable’

2. Scenario Two:
    a. Create a Contact
    b. Create a GE record with above Contact and Status as ‘Current’ with ‘Grade Level’ as ‘Prekindergarten’
    c. Ensure Current Grade Level on Contact is populated as ‘Prekindergarten’
    d. Turn off the ‘GRER_SyncGradeLevel_TDTM’ by unchecking the ‘Active’ box for the TDTM record in the ‘Trigger handlers’ tab
    e. Create a GE record with above Contact and Status as ‘Current’ with ‘Grade Level’ as ‘First Grade’
    f. Turn back the Trigger handler record on
    g. Delete the GE record from step ‘b’
    h. The Current Grade Level on the Contact should now change to ‘First Grade’

3. Scenario Three:
    a. Create a Contact
    b. Create a GE record with above Contact and Status as ‘Current’ with ‘Grade Level’ as ‘Prekindergarten’
    c. Ensure Current Grade Level on Contact is populated as ‘Prekindergarten’
    d. Turn off the ‘GRER_SyncGradeLevel_TDTM’ by unchecking the ‘Active’ box for the TDTM record in the ‘Trigger handlers’ tab
    e. Create 1 GE record with above Contact and Status as ‘Current’ with ‘Grade Level’ as ‘First Grade’
    f. Create another GE record with above Contact and Status as ‘Current’ with ‘Grade Level’ as ‘Second Grade’
    g. Turn back the Trigger handler record on
    h. Delete the GE record from step ‘b’
    i. You should receive the error ‘We’ve found more than one Grade Enrollment record specified as Current. Review your Grade Enrollment records and ensure that only one is set as Current before changing the status of this record.’

*Note* : In order to test the multiple “Current” Grade Level edge case, you’d have to turn off the trigger handler and setup the data to have multiple “Current” Grade Enrollments and then run through the above scenario but verify that the Current Grade Level is set to a current Grade Enrollment.

### CODE REVIEW

File: src/classes/GRER_SyncGradeLevel_TDTM.cls [hunk 1]
```diff
@@ -48,60 +48,94 @@ global with sharing class GRER_SyncGradeLevel_TDTM extends hed.TDTM_Runnable {
...
         // Turn off After Update for CON_CurrentGradeLevel_TDTM
         GRER_SyncGradeLevel_TDTM.turnOffGradeEnrollmentCreation();
-                                       
+
         //After INSERT
         if (triggerAction == hed.TDTM_Runnable.Action.AfterInsert) {
             if (listNew != null && listNew.size() > 0) {
                 List<Grade_Enrollment__c> newGradeEnrols = (List<Grade_Enrollment__c>)listNew;
-                selectGradeEnrolAndContactsToUpdate(newGradeEnrols, NULL, dmlWrapper, triggerAction);
+                selectGradeEnrolAndContactsToUpdate(newGradeEnrols, NULL, NULL, dmlWrapper, triggerAction);
```
[the remark this answers is not in the document]
Comment (skosgi1) [reply src/classes/GRER_SyncGradeLevel_TDTM.cls]: Resolving the thread. @spinkelman to reach out to Ryan and create a WI if needed.

**Comment by spinkelman** [comment]
Passes QA - Merging
Will Close WI
